frontend/src/pages/Graphing/components/Graph.tsx

```diff
@@ -359,37 +364,102 @@ const Graph = ({
 	onExpand,
 	onEdit,
 	setTimeRange,
+	selectedPreset,
 }: ChartProps<ViewConfig>) => {
 	const [graphHover, setGraphHover] = useState(false)
 	const queriedBucketCount = bucketByKey !== undefined ? bucketCount : 1
 	const showMenu =
 		onDelete !== undefined || onExpand !== undefined || onEdit !== undefined
 
-	const { data: metrics, loading: metricsLoading } = useGetMetricsQuery({
-		variables: {
-			product_type: productType,
-			project_id: projectId,
-			params: {
-				date_range: {
-					start_date: moment(startDate)
-						.startOf('minute')
-						.format(TIME_FORMAT),
-					end_date: moment(endDate)
-						.startOf('minute')
-						.format(TIME_FORMAT),
+	const pollTimeout = useRef<number>()
+	const [pollInterval, setPollInterval] = useState<number>(0)
```

Comment:
could you do this with the apollo pollInterval setting or did you need more flexibility? https://www.apollographql.com/docs/react/data/queries/#polling

Reply:
I think more flexibility was needed - the issue I was running into was the request would keep requesting the same start and end times.

So the polling function is actually just "rebasing" the start and end times, which then triggers a new request. Happy to look back into this though if we feel it would be better